### ISSUE TITLE
NoSuchMethodError: NoSuchMethodError: The method 'forEach' was called on null.

### ISSUE THREAD

**Issue by Rayv1** [aws-amplify/amplify-flutter#156]
Hi,

Note: these bugs are not happening, when running in debug Mode or release Mode on a  Software / Hardware Device. Its atm only happening when building an APK  / AppBundle File for Android with release flag

First, when i try to Login a User, the App is complete crushing. i have no clue how to get logs or a stracktrace from it, as its not happening in debug  mode.

Second Error happens when trying to signUp a new User.

```
NoSuchMethodError: NoSuchMethodError: The method 'forEach' was called on null.
Receiver: null
Tried calling: forEach(Closure: (dynamic, dynamic) => Set<Set<dynamic>>)
  File "object_patch.dart", line 51, in Object.noSuchMethod
  File "method_channel_auth_cognito.dart", line 330, in AmplifyAuthCognitoMethodChannel._throwError
  File "method_channel_auth_cognito.dart", line 40, in AmplifyAuthCognitoMethodChannel.signUp
  File "<asynchronous suspension>"
  File "amplify_auth_cognito.dart", line 45, in AmplifyAuthCognito.signUp
  File "<asynchronous suspension>"
  File "authentication_service.dart", line 207, in AuthenticationService.signUpUser
  File "<asynchronous suspension>"
  File "login_view_model.dart", line 197, in LoginRegisterViewModel.signUp
  File "<asynchronous suspension>"
  File "login_view.dart", line 349, in LoginRegisterUserView.build.<fn>.<fn>
  File "<asynchronous suspension>"
  File "unparsed"
```

Steps to reproduce: 
1. flutter build apk --release (--debug is working)
2. flutter install

**Comment by Rayv1** [comment]
okay it seems to be fixed when running:
flutter build apk --release **--no-shrink**

Maybe this is something you should know.

**Comment by haverchuck** [comment]
@Rayv1 It is good to hear you are unblocked, and yes this is definitely something we should look into. Thanks for posting this issue.

**Comment by gaurav618618** [comment]
I am also getting the same issue when i build apk through android studio and deploy it . I am facing issues with only **Signup** function and everything else in **Auth** is working fine for me . **Signup** works fine in debug mode that's why i was so confused . After deploying apk through android studio i got the following error in logcat when apk calls signup fucntion.

```
2020-10-28 07:05:27.554 14035-14035/? E/MethodChannel#com.amazonaws.amplify/auth_cognito: Failed to handle method call
    e.d: null cannot be cast to non-null type java.lang.reflect.Method
        at com.amazonaws.amplify.amplify_auth_cognito.types.FlutterSignUpRequest.a(Unknown Source:203)
        at com.amazonaws.amplify.amplify_auth_cognito.types.FlutterSignUpRequest.<init>(Unknown Source:96)
        at com.amazonaws.amplify.amplify_auth_cognito.AuthCognito.i(Unknown Source:10)
        at com.amazonaws.amplify.amplify_auth_cognito.AuthCognito.a(Unknown Source:150)
        at d.a.b.a.j$a.a(Unknown Source:17)
        at io.flutter.embedding.engine.e.b.a(Unknown Source:57)
        at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(Unknown Source:4)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:336)
        at android.os.Looper.loop(Looper.java:174)
        at android.app.ActivityThread.main(ActivityThread.java:7397)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:935)
2020-10-28 07:05:27.556 14035-14063/? E/flutter: [ERROR:flutter/lib/ui/ui_dart_state.cc(166)] Unhandled Exception: NoSuchMethodError: The method 'forEach' was called on null.
    Receiver: null
    Tried calling: forEach(Closure: (dynamic, dynamic) => Set<Set<dynamic>>)
    #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51)
    #1      AmplifyAuthCognitoMethodChannel._throwError (package:amplify_auth_cognito/method_channel_auth_cognito.dart:330)
    #2      AmplifyAuthCognitoMethodChannel.signUp (package:amplify_auth_cognito/method_channel_auth_cognito.dart:40)
    <asynchronous suspension>
    #3      AmplifyAuthCognito.signUp (package:amplify_auth_cognito/amplify_auth_cognito.dart:45)
    <asynchronous suspension>
    #4      _State._registerUser (package:confab_app/SignUp_Screen.dart:36)
    <asynchronous suspension>
```

**Comment by jamesonwilliams** [comment]
See @gaurav618618's related StackOverflow post, here:

https://stackoverflow.com/questions/64565537/aws-amplify-flutters-auth-signup-function-is-not-working-when-deployed-via-andr

**Comment by jamesonwilliams** [comment]
> Unhandled Exception: NoSuchMethodError: The method 'forEach' was called on null

I think one of two things is happening here:
1. R8 is stripping out something it shouldn't be. You need to update ProGuard rules configuration. R8 is usually only run on release builds, so that would explain the difference between the debug/release.
2. (Less likely) You're running this on an older API level, where `forEach` isn't available. `forEach` is only available from API 24+.

**Comment by Rayv1** [comment]
@jamesonwilliams can u provide those ProGuard Rules for amplify?

**Comment by jamesonwilliams** [comment]
> @jamesonwilliams can u provide those ProGuard Rules for amplify-flutter?

Haven't had time to look at this in depth, but here's something to unblock you. Credit to @momo1:

https://github.com/aws-amplify/amplify-flutter/issues/286#issuecomment-753545450

```
-keep class com.amazonaws.** { *; }
-keep class com.amplifyframework.** { *; }
```

**Comment by Amplifiyer** [comment]
This is released today in version 0.0.2-dev.2. Let us know if you are still facing any issues. Closing this.
